src/app/Shared/Services/Api.service.tsx

```diff
@@ -690,51 +748,64 @@ export class ApiService {
       });
   }
 
-  uploadRecording(file: File, labels: Object, signal?: AbortSignal): Observable<string> {
-    window.onbeforeunload = () => true;
-    return this.login.getHeaders().pipe(
-      concatMap((headers) => {
-        const body = new window.FormData();
-        body.append('recording', file);
-        body.append('labels', JSON.stringify(labels));
+  uploadRecording(
+    file: File,
+    labels: Object,
+    onUploadProgress?: (progress: number) => void,
+    abortSignal?: Observable<void>
+  ): Observable<string> {
+    window.onbeforeunload = (event: BeforeUnloadEvent) => event.preventDefault();
```

Comment:
Should the other upload functions have this too?

Reply:
Missed that too. Added back :D